### PR TITLE
Upgrade image dependency + Bump version

### PR DESCRIPTION
**Pull request by actioninja**
Upgrades the image dependency to 0.24 + bumps the version.

Annoyingly, image changed its apis slightly without bumping major semver.

### CODE REVIEW

**Review by ZeWaka — CHANGES_REQUESTED** [review]
rustfmt

**Comment by actioninja** [comment]
going to fix the clippy issues because they're pretty trivial as well

**Comment by actioninja** [comment]
Changes made, should be good to to rerun CI

**Comment by actioninja** [comment]
I should just setup a rustfmt githook

File: src/dmi/icon.rs [hunk 1]
```diff
@@ -5,7 +5,9 @@ use super::RawDmi;
 use image::imageops;
 use image::GenericImageView;
 use std::collections::HashMap;
+use std::fmt::Write as _;
```
Comment (ZeWaka) [inline src/dmi/icon.rs:8]: i don't get the need for all of this

Comment (actioninja) [reply src/dmi/icon.rs:8]: it's an optimization over calling format! and then calling push_str that was flagged by clippy and causing the CI to fail. Avoids an unneeded allocation.

**Comment by ZeWaka** [comment]
closing in favor of #5

**Pull request closed without merging**